File: modules/spring-web/testdata/java/inspection/initBinderReturnType/src/com/InitBinderController.java
package com.example.springcoredemo.web;

import org.springframework.web.bind.WebDataBinder;
import org.springframework.web.bind.annotation.*;

@RestController
public class InitBinderController {
    
    @InitBinder("valid")
    public void validBinderMethod(WebDataBinder dataBinder) {}
    
    @InitBinder("invalid")
    public String invalidBinderMethod(WebDataBinder dataBinder) {
        return "";
    }
    
}

File: modules/spring-web/testdata/kotlin/inspection/initBinderReturnType/src/com/InitBinderController.kt
import org.springframework.web.bind.WebDataBinder
import org.springframework.web.bind.annotation.*


@RestController
class InitBinderController {
    @InitBinder("valid")
    fun validBinderMethod(dataBinder: WebDataBinder?) {
    }

    @InitBinder("invalid")
    fun invalidBinderMethod(dataBinder: WebDataBinder?): String {
        return ""
    }
}